pkg/artifactory/resource_artifactory_remote_repository_test.go

```diff
@@ -166,6 +166,10 @@ func resourceRemoteRepositoryCheckDestroy(id string) func(*terraform.State) erro
 
 		_, resp, err := client.V1.Repositories.GetRemote(context.Background(), rs.Primary.ID)
 
+		if resp == nil {
```

Comment:
Check `err` first. If you check error first you probably never need to check if `resp` is `nil`.